packages/node/src/subcommands/reindex.service.ts

```diff
@@ -50,12 +45,21 @@ export class ReindexService {
       throw new Error('Schema does not exist.');
     }
     await this.initDbSchema();
-
     this.metadataRepo = MetadataFactory(this.sequelize, this.schema);
+  }
 
-    this.unfinalizedBlocksService.init(this.metadataRepo, () =>
-      Promise.resolve(),
+  async getTargetHeightWithUnfinalizedBlocks(inputHeight): Promise<number> {
+    (this.unfinalizedBlocksService as any).metadataRepo = this.metadataRepo;
+    const unfinalizedBlocks =
+      await this.unfinalizedBlocksService.getMetadataUnfinalizedBlocks();
+    const bestBlocks = unfinalizedBlocks.filter(
+      ([bestBlockHeight]) => Number(bestBlockHeight) <= inputHeight,
     );
+    if (bestBlocks.length === 0) {
+      return inputHeight;
+    }
+    const [firstBestBlock] = bestBlocks[0];
+    return firstBestBlock;
```

Comment:
If there are unfinalized blocks we should still consider the inputHeight

```suggestion
    return Math.min(inputHeight, firstBestBlock);
```